lib/cluster/index.js

```diff
@@ -57,7 +57,7 @@ function Cluster(startupNodes, options) {
 
   var _this = this;
   this.connectionPool.on('-node', function (redis) {
-    if (_this.subscriber === redis) {
+    if (_this.status === 'ready' && _this.subscriber === redis) {
```

Comment:
Enum may not be able to avoid typos IMO. For example, I may wrongly write `Redis.READY` as `Redis.REEDY`, and the latter will be an `undefined`, no exception would be thrown.